mcpserver/tools.go

```diff
@@ -0,0 +1,1036 @@
+// Copyright (c) 2023-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package mcpserver
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/mark3labs/mcp-go/mcp"
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/shared/mlog"
+)
+
+// MattermostToolProvider implements Mattermost operations for MCP tools
+type MattermostToolProvider struct {
```

Comment:
Not really a provider if there isn't a generic interface. Do we need this abstraction?